spec/Appendix B -- Grammar Summary.md

```diff
@@ -43,7 +43,14 @@ Token ::
 - FloatValue
 - StringValue
 
-Punctuator :: one of ! $ & ( ) ... : = @ [ ] { | }
+Punctuator ::
+
+- DotPunctuator
+- OtherPunctuator
```

Comment:
Out of curiosity, why do we even need that lexical rule?  Most of the parsers I've bumped into do not have a `Punctuator` token and just use the underlying character directly. `Punctuator` doesn't seem to be used by any grammar rule either?

Reply:
It's one of the `Token`s supported by the lexer; lex the input stream into tokens, then parse the tokens into an AST.